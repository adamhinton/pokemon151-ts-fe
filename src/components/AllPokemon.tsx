// This takes an array of Pokemon objects and returns an individual card (SinglePokemon.js) for each object
import React, { ReactElement } from "react";
import { PokemonData } from "../data";
import SinglePokemon from "./SinglePokemon";

interface Props {
  pokemonData: PokemonData;
  isShiny: Boolean;
}

const AllPokemon = (props: Props): ReactElement => {
  const { pokemonData, isShiny } = props;
  return (
    <main className="all-pokemon">
      {pokemonData.map((poke) => {
        return <SinglePokemon poke={poke} key={poke.name} isShiny={isShiny} />;
      })}
    </main>
  );
};
export default AllPokemon;
