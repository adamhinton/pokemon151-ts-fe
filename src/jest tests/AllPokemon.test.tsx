import { render, screen } from "@testing-library/react";
import AllPokemon from "../components/AllPokemon";
import pokemonData, { PokemonData } from "../data";

jest.mock("../data", (): PokemonData => {
  return [
    {
      id: 1,
      num: "001",
      name: "Testname123",
      type: ["TestType1", "TestType2"],
      height: "0.0001m",
      weight: "0.0001kg",
      dexEntry: "This is a test dex entry",
    },
    {
      id: 2,
      num: "002",
      name: "Testname2",
      type: ["TestType2", "TestType3"],
      height: "0.0001m",
      weight: "0.0001kg",
      dexEntry: "This is a test dex entry",
    },
  ];
});

test("[1] Renders without errors", () => {
  render(<AllPokemon pokemonData={pokemonData} isShiny={true} />);
});

test("[2] All pokemon objects appear on screen", () => {
  render(<AllPokemon pokemonData={pokemonData} isShiny={true} />);

  const pokemonList = screen.getAllByTestId("pokemon-card");

  expect(pokemonList.length).toBe(2);
});
