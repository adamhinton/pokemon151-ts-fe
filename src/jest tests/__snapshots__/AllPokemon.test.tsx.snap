// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[4] Matches snapshot from 9.17.22 1`] = `
<main
  className="all-pokemon"
>
  <div>
    This is a mocked SinglePokemon Component
  </div>
  <div>
    This is a mocked SinglePokemon Component
  </div>
  <div>
    This is a mocked SinglePokemon Component
  </div>
</main>
`;
